posts/zzz_DO_NOT_EDIT_document_yo.../document_your_universe_with_dverse.qmd

```diff
@@ -0,0 +1,186 @@
+  collapse = TRUE
+)
+```
+
+<!--------------- post begins here ----------------->
+
+This article shows how to create a global search for any collection of R
+packages, i.e. a universe. Your universe may be on CRAN, large, and popular like
+the [tidyverse](https://tidyverse.org/) or
+[pharmaverse](https://pharmaverse.org/). But it may also be not on CRAN, small,
+company-specific or personal.
+
+The example here uses the 'admiralverse' -- a toy universe defined as every
+package on CRAN today (2024-12-07), with a website, and a name starting with
+'admiral'.
+
+You need:
+
+* the {dverse} package, 
+* each package in the toy 'admiralverse,
+* {dplyr} to manipulate the output of {dverse}, and 
+* {DT} to create a searchable table.
+
+```{r packages}
+library(dverse)
+library(admiral)
+library(admiraldev)
+library(admiralonco)
+library(admiralophtha)
+library(admiralpeds)
+library(admiralvaccine)
+library(dplyr)
+library(DT)
+```
+
+<details><summary>Versions</summary>
+
+```{r versions}
+packageVersion("dverse")
+packageVersion("admiral")
+packageVersion("admiraldev")
+packageVersion("admiralonco")
+packageVersion("admiralophtha")
+packageVersion("admiralpeds")
+packageVersion("admiralvaccine")
+packageVersion("dplyr")
+packageVersion("DT")
+```
+
+</details>
+
+dverse::document_universe() takes the names of the packages in the
+'admiralverse', and a template pointing to each help file (topic) in each
+package.
+
+```{r docs}
+admiralverse <- c(
+  "admiral",
+  "admiraldev",
+  "admiralonco",
+  "admiralophtha",
+  "admiralpeds",
+  "admiralvaccine"
+)
+# For example: https://pharmaverse.github.io/admiral/reference/queries.html
+template <- "https://pharmaverse.github.io/{package}/reference/{topic}.html"
+
+docs <- dverse::document_universe(admiralverse, template)
+docs
+```
+
+You may need to exclude some topics, such as reexported objects from other
+packages. Because topics outside your universe yield broken links, you can
+exclude them with dverse::is_online().
+
+```{r online}
+docs |> filter(!dverse::is_online(topic))
+
+online <- docs |> filter(dverse::is_online(topic))
+```
+
+Consider the different kinds of documentation in your universe.
+
+```{r kinds}
+online |> count(type, keyword)
+```
+
+Here it makes sense to separate vignettes, datasets, and public functions.
+
+### Vignettes
+
+```{r vignettes}
+vignettes <- online |> filter(type == "vignette")
+
+vignettes |>
+  distinct(topic, title, package) |>
+  datatable(escape = FALSE)
+```
+
+#### Don't you see what you expect?
+
+* This works with vignettes but not articles, so usethis::use_vignette() not
+usethis::use_article().
+
+* When you install the packages in your universe ensure to install the vignettes.
+Vignettes install by default when you install your packages from CRAN with
+`install.packages()` or pak::pak(). But if you install them from source with
+devtools::install() or remotes::install_github(), then you need `build_vignettes
+= TRUE`.
+
+### Datasets
+
```

Comment:
Could you add some context to this chapter?

Reply:
Thanks! I added more context in all three of the problematic sections: vignettes, datasets, and functions.

![image](https://github.com/user-attachments/assets/00de043e-48e5-45b2-83b9-975a9157347b)

...

![image](https://github.com/user-attachments/assets/15cce03b-c928-4c60-bb85-ad7ec7aebd66)

...

![image](https://github.com/user-attachments/assets/cfaee0d4-6bd2-4eef-82b8-2d9b54c603d8)